UnityProject/Assets/Scripts/Systems/StationObjectives/StationObjectiveManager.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Text;
+using System.Linq;
+using DiscordWebhook;
+using DatabaseAPI;
+using Messages.Server.LocalGuiMessages;
+using Strings;
+using Managers;
+using Random = UnityEngine.Random;
+
+namespace StationObjectives
+{
+	public class StationObjectiveManager : MonoBehaviour
+	{
+		public GameManager gameManager;
+
+		[SerializeField]
+		[Tooltip("Stores all station objective data.")]
+		private StationObjectiveData stationObjectiveData = null;
+
+		public static StationObjectiveManager Instance;
+
+		[NonSerialized] public StationObjective activeObjective;
+
+		private List<Vector2> asteroidLocations = new List<Vector2>();
+
+		private void Awake()
+		{
+			if (Instance == null)
+			{
+				Instance = this;
+			}
+			else
+			{
+				Destroy(gameObject);
+			}
+		}
+		void OnEnable()
+		{
+			EventManager.AddHandler(EVENT.RoundEnded, ResetObjectives);
+		}
+		void OnDisable()
+		{
+			EventManager.RemoveHandler(EVENT.RoundEnded, ResetObjectives);
+		}
+
+		//public ;
+
+		public void RemindStation()
+		{
+			GameManager.Instance.CentComm.MakeCommandReport(StationObjectiveReport(),CentComm.UpdateSound.Notice);
+		}
+		private string StationObjectiveReport()
+		{
+			var report = new StringBuilder(activeObjective.Description);
+			report.Replace("STATIONNAME", MatrixManager.MainStationMatrix.GameObject.scene.name);
+
+			foreach (var location in asteroidLocations)
+			{
+				report.AppendFormat(" <size=24>{0}</size> ", Vector2Int.RoundToInt(location));
+			}
+
+			return report.ToString();
+		}
+
+		public void ServerChooseObjective()
+		{
+			foreach (var body in gameManager.SpaceBodies)
```

Comment:
I don't understand what is the asteroids location doing here

Reply:
for the printout at roundstart, it records a few asteroid coords

Reply:
that's only important for the ship resources objective, isn't it?

Reply:
good point, i'll see if i can move it to the ship resources file